arangod/RestHandler/RestVocbaseBaseHandler.cpp

```diff
@@ -595,6 +596,10 @@ RestVocbaseBaseHandler::createTransaction(
     if (isFollower) {
       tmp->addHint(transaction::Hints::Hint::IS_FOLLOWER_TRX);
     }
+    if (isDBServer) {
+      // set username from request
+      tmp->setUsername(_request->value(StaticStrings::UserString));
+    }
```

Comment:
Just curious: Why is this only necessary on a DBServer?

Reply:
This is only used in context of the new metrics. And the new metrics are based on shards, which only exist on DB servers.